pkg/e2e/infra/utils.go

```diff
@@ -319,3 +319,32 @@ func getScaleClient() (scale.ScalesGetter, error) {
 	}
 	return scaleClient, nil
 }
+
+// nodesSnapShot logs the state of all the nodes in the cluster
+func nodesSnapShot(client runtimeclient.Client) error {
+	nodeList, err := getNodeList(client)
+	if err != nil {
+		return fmt.Errorf("error calling getMachineSetList: %v", err)
+	}
+	for key := range nodeList.Items {
+		glog.Infof("Node %q. Ready: %t. Unschedulable: %t", nodeList.Items[key].Name, e2e.IsNodeReady(&nodeList.Items[key]), nodeList.Items[key].Spec.Unschedulable)
+	}
+	return nil
+}
+
+// getNodeList returns a nodeList object
+func getNodeList(client runtimeclient.Client) (*corev1.NodeList, error) {
```

Comment:
As per some of the other recent changes it we don't need the MetaType info from NodeList then return a slice of nodes.